src/crypto/sha256.cpp

```diff
@@ -623,6 +638,35 @@ std::string SHA256AutoDetect()
 #endif
 #endif
 
+#if defined(ENABLE_ARM_SHANI) && !defined(BUILD_BITCOIN_INTERNAL)
+    bool have_arm_shani = false;
+
+#if defined(__linux__)
+#if defined(__arm__) // 32-bit
+    if (getauxval(AT_HWCAP2) & HWCAP2_SHA2) {
+        have_arm_shani = true;
+    }
+#endif
+#if defined(__aarch64__) // 64-bit
+    if (getauxval(AT_HWCAP) & HWCAP_SHA2) {
+        have_arm_shani = true;
+    }
+#endif
+#endif
+
+#if defined(__APPLE__)
```

Comment:
Can you use `MAC_OSX` here to match our other macOS defines.

Reply:
Addressed in 853b4c97995720e56500c292b7d8516f04064f67